backend/de.metas.business/src/main/java/de/metas/invoice/invoiceProcessingServiceCompany/InvoiceProcessingServiceCompanyConfigMap.java

```diff
@@ -68,15 +74,40 @@ public InvoiceProcessingServiceCompanyConfigMap(@NonNull final List<InvoiceProce
 		}
 	}
 
+	/**
+	 * @return if there is an {@link InvoiceProcessingServiceCompanyConfig} with the given {@code customerId} (the one for which a service-company handles payments) at the given {@code validFrom}, then it is returned.<br>
+	 * If there is an older config for the given {@code customerId}, but a more recent config which does not have the given {@code customerId},
+	 * then {@link Optional#empty()} is returned.
+	 */
+	@NonNull
 	public Optional<InvoiceProcessingServiceCompanyConfig> getByCustomerIdAndDate(@NonNull final BPartnerId customerId, @NonNull final ZonedDateTime validFrom)
 	{
-		final ImmutableList<InvoiceProcessingServiceCompanyConfig> configs = bpartnersToConfigsSorted.get(customerId);
-		for (int i = configs.size() - 1; i >= 0; i--)
+		final ImmutableList<InvoiceProcessingServiceCompanyConfig> configsForCustomers = bpartnersToConfigsSorted.get(customerId);
+		for (int i = configsForCustomers.size() - 1; i >= 0; i--)
```

Comment:
why iterate from last to first? how are the bpartnersToConfigsSorted?

Reply:
the values are sorted by validFrom, index 0 will date the record with the oldest (the most far in the past) validFrom and the last index will have the most recent validFrom